client/my-sites/plans-features-main/index.jsx

```diff
@@ -533,7 +518,9 @@ export default connect(
 		const currentPlan = getSitePlan( state, siteId );
 		const sitePlanSlug = currentPlan?.product_slug;
 		const eligibleForWpcomMonthlyPlans =
-			isWpComFreePlan( sitePlanSlug ) || isWpComMonthlyPlan( sitePlanSlug );
+			( isAtomicSite( state, siteId ) && sitePlanSlug === 'jetpack_free' ) ||
+			isWpComFreePlan( sitePlanSlug ) ||
+			isWpComMonthlyPlan( sitePlanSlug );
```

Comment:
@pablinos To your point about fewer conditionals = less fragmentation (pd2C1Z-Q-p2#comment-21), is there anything we can do to consolidate these multi-conditionals in the short-term? E.g. add them to a helper or at a higher level? I tried [adding a helper here](https://github.com/Automattic/wp-calypso/pull/53196/files).